internal/policy/container/has_modified_files.go

```diff
@@ -24,17 +26,26 @@ var _ check.Check = &HasModifiedFilesCheck{}
 // modified in subsequent layers.
 type HasModifiedFilesCheck struct{}
 
+type (
+	layerFiles        = map[string][]string
+	layerPackages     = map[string]map[string]struct{}
+	layerPackageFiles = map[string]map[string]string
+)
+
 type packageFilesRef struct {
-	// LayerFiles contains a slice of files modified by layer.
-	LayerFiles [][]string
-	// PackagesFiles is a map of modified files. the anonymous struct
-	// here is just used to allow us to search package files by name
-	// as the key of this map, and doesn't have any value.
-	PackageFiles map[string]struct{}
+	// LayerFiles contains a slice of files created/modified in layer
+	LayerFiles layerFiles
+	// LayerPackages is a map of a map of packages for each layer
+	LayerPackages layerPackages
+	// LayerPackageFiles maps files to a package name-version
+	LayerPackageFiles layerPackageFiles
+	// Layers is the list of layers in the image
+	Layers []string
```

Comment:
I'm not sure this is any more clear with these type definitions. I think what I expected to see instead was something like...

```go
type layerFiles struct {
    id string
    files []string
}
```

Feel free to disregard, though if you do, I'm not sure if the type definitions as written here add value, so you can remove them from my point of view. Up to you.

Reply:
At this point, it's an implementation detail. It will not affect the outside world, so if we want to change it in the future, we can. For now, this is how my brain is interpreting the data structure. I know there is a (probably) a better way, but I don't really want to focus on that atm